internal/plugin.go

```diff
@@ -376,6 +395,105 @@ func NewLuaPlugin(content, path string, manager *Manager) (*LuaPlugin, error) {
 	return source, nil
 }
 
+// NewLuaPlugin creates a new LuaPlugin instance from the specified directory path.
+// The plugin directory must meet one of the following conditions:
+// - The directory must contain a metadata.lua file and a hooks directory that includes all must be implemented hook functions.
+// - The directory contain a main.lua file that defines the plugin object and all hook functions.
+func NewLuaPlugin(pluginDirPath string, manager *Manager) (*LuaPlugin, error) {
+	vm := luai.NewLuaVM()
+
+	if err := vm.Prepare(&luai.PrepareOptions{
+		Config: manager.Config,
+	}); err != nil {
+		return nil, err
+	}
+
+	mainPath := filepath.Join(pluginDirPath, "main.lua")
+	// main.lua first
+	if util.FileExists(mainPath) {
+		vm.LimitPackagePath(filepath.Join(pluginDirPath, "?.lua"))
+		if err := vm.Instance.DoFile(mainPath); err != nil {
+			return nil, err
+		}
+	} else {
+		// Limit package search scope, hooks directory search priority is higher than lib directory
+		hookPath := filepath.Join(pluginDirPath, "hooks", "?.lua")
+		libPath := filepath.Join(pluginDirPath, "lib", "?.lua")
+		vm.LimitPackagePath(hookPath, libPath)
+
+		// load metadata file
+		metadataPath := filepath.Join(pluginDirPath, "metadata.lua")
+		if !util.FileExists(metadataPath) {
+			return nil, fmt.Errorf("plugin invalid, metadata file not found")
+		}
+
+		if err := vm.Instance.DoFile(metadataPath); err != nil {
+			return nil, fmt.Errorf("failed to load meatadata file, %w", err)
+		}
+
+		// load hook func files
+		for _, hf := range HookFuncs {
+			hp := filepath.Join(pluginDirPath, "hooks", hf.Filename+".lua")
```

Comment:
can we access these file when we need?
when we exeucute `use`, we won't execute `XXXInstall` hooks

Reply:
Of course, I originally planned to do this, but considering the need to be compatible with previous plug-ins(installed plugins by user), because previous plug-ins and hooks call each other, if we don't do this, it will be a **break change**.

Reply:
Oh no, installed plugins are not affected because they are all in one file. I will change it later!

Reply:
Oops, After modding it, I noticed that we used the PLUGIN global variable to mount the hook function. If we load it when calling, because it is a function on the object, there is no way to restrict the user from calling other hook functions internally when developing.